### ISSUE TITLE
Custom dataset performance issue

### ISSUE THREAD

**Issue by neverstoplearn** [hukaixuan19970627/yolov5_obb#263]

**Comment by hukaixuan19970627** [comment]
![image](https://user-images.githubusercontent.com/72599120/161373129-f8dc6488-47f5-4afb-8ee8-9ba959f0b496.png)
将上述选定的文件内容截图发一下，我看看训练情况

**Comment by neverstoplearn** [comment]
"hpy.yaml":
lr0: 0.0032
lrf: 0.12
momentum: 0.843
weight_decay: 0.00036
warmup_epochs: 2.0
warmup_momentum: 0.5
warmup_bias_lr: 0.05
box: 0.0296
cls: 0.243
cls_pw: 0.631
obj: 0.301
obj_pw: 0.911
angle: 0.03
angle_pw: 0.05
iou_t: 0.2
anchor_t: 2.91
fl_gamma: 0.0
hsv_h: 0.0138
hsv_s: 0.664
hsv_v: 0.464
degrees: 0
translate: 0
scale: 0
shear: 0
perspective: 0.0
flipud: 0
fliplr: 0
mosaic: 0
mixup: 0
copy_paste: 0.0

"opt.yaml":
weights: ./weights/yolov5s.pt
cfg: ./models/yolov5s.yaml
data: ./data/dotav15_poly.yaml
hyp: ./data/hyps/hyp.finetune.yaml
epochs: 300
batch_size: 4
imgsz: 640
rect: false
resume: false
nosave: false
noval: false
noautoanchor: false
evolve: null
bucket: ''
cache: null
image_weights: false
device: ''
multi_scale: false
single_cls: false
adam: false
sync_bn: false
workers: 8
project: runs/train
name: exp
exist_ok: false
quad: false
linear_lr: false
label_smoothing: 0.0
patience: 100
begin_val: 300
end_mosaic: 0
freeze: 0
save_period: -1
local_rank: -1
entity: null
upload_dataset: false
bbox_interval: -1
artifact_alias: latest
save_dir: runs/train/exp25

**Comment by neverstoplearn** [comment]
"hpy.yaml":
lr0: 0.0032
lrf: 0.12
momentum: 0.843
weight_decay: 0.00036
warmup_epochs: 2.0
warmup_momentum: 0.5
warmup_bias_lr: 0.05
box: 0.0296
cls: 0.243
cls_pw: 0.631
obj: 0.301
obj_pw: 0.911
angle: 0.03
angle_pw: 0.05
iou_t: 0.2
anchor_t: 2.91
fl_gamma: 0.0
hsv_h: 0.0138
hsv_s: 0.664
hsv_v: 0.464
degrees: 0
translate: 0
scale: 0
shear: 0
perspective: 0.0
flipud: 0
fliplr: 0
mosaic: 0
mixup: 0
copy_paste: 0.0

"opt.yaml":
weights: ./weights/yolov5s.pt
cfg: ./models/yolov5s.yaml
data: ./data/dotav15_poly.yaml
hyp: ./data/hyps/hyp.finetune.yaml
epochs: 300
batch_size: 4
imgsz: 640
rect: false
resume: false
nosave: false
noval: false
noautoanchor: false
evolve: null
bucket: ''
cache: null
image_weights: false
device: ''
multi_scale: false
single_cls: false
adam: false
sync_bn: false
workers: 8
project: runs/train
name: exp
exist_ok: false
quad: false
linear_lr: false
label_smoothing: 0.0
patience: 100
begin_val: 300
end_mosaic: 0
freeze: 0
save_period: -1
local_rank: -1
entity: null
upload_dataset: false
bbox_interval: -1
artifact_alias: latest
save_dir: runs/train/exp25

**Comment by neverstoplearn** [comment]
![labels](https://user-images.githubusercontent.com/44988584/161374329-6ec50a6e-a46d-466f-b6b0-5719bb50ed85.jpg)

**Comment by neverstoplearn** [comment]
![labels_correlogram](https://user-images.githubusercontent.com/44988584/161374333-566d82f5-fe5c-495d-8c04-87b8b6a3c693.jpg)

**Comment by neverstoplearn** [comment]
![2022-04-02 16-25-58屏幕截图](https://user-images.githubusercontent.com/44988584/161374453-38640af8-a4c0-4793-a663-41d20c5e3814.png)

**Comment by neverstoplearn** [comment]
如上图bacth0所示，大部分标签是正常的，但是经营范围后面的部分是标注了的  但是没有显示 通过你提供的DOTA_devkit_YOLO转换时保存的图片是有的，因为完整的数据集涉及到身份证 户口本等隐私数据所以没有提供完整的trainbatch，抱歉。
![yyzx4-(854)_](https://user-images.githubusercontent.com/44988584/161374539-bcc5dee6-502f-4192-b98e-49bcd462d39c.jpg)

**Comment by hukaixuan19970627** [comment]
确实奇怪，代码里是会删除部分目标，不过根据处理逻辑只会删除长边短边均小于5的极小目标。
你先单独用这张图构建一个数据集再测试一下，我更倾向于认为这张图的部分目标在txt中就已经没了

**Comment by hukaixuan19970627** [comment]
等一下，你这个代码似乎不是最新版本的代码，是老版本的吧。建议使用最新版本的代码，训练效率提升很明显，精度也更高，而且不需要使用DOTA_devkit_YOLO来转换labels，代码里会自动完成对DOTA format label的转换工作

**Comment by neverstoplearn** [comment]
> 等一下，你这个代码似乎不是最新版本的代码，是老版本的吧。建议使用最新版本的代码，训练效率提升很明显，精度也更高，而且不需要使用DOTA_devkit_YOLO来转换labels，代码里会自动完成对DOTA format label的转换工作（不过要是你的系统是windows的话那还是继续用老版本吧，新版本代码需要编译cuda代码，windows容易报错）
我这个数据集之前是用DBNet分割方法作的，所以标注的格式是icdar格式，即x1,y1,x2,y2,x3,y3,x4,y4. 意思是最新的代码直接读入这在格式的标签吗？不需要转为x y w h theta吗？

**Comment by neverstoplearn** [comment]
> 等一下，你这个代码似乎不是最新版本的代码，是老版本的吧。建议使用最新版本的代码，训练效率提升很明显，精度也更高，而且不需要使用DOTA_devkit_YOLO来转换labels，代码里会自动完成对DOTA format label的转换工作（不过要是你的系统是windows的话那还是继续用老版本吧，新版本代码需要编译cuda代码，windows容易报错）

看到了 不需要转换了
